File: cis_interface/examples/rpcFib/src/rpcFibCliPar.py
from __future__ import print_function
import sys
from cis_interface.interface.PsiInterface import PsiRpcClient


def fibClient(args):
    
    iterations = int(args[0])
    print('Hello from Python rpcFibCliPar: iterations = %d' % iterations)

    # Create RPC connection with server
    rpc = PsiRpcClient("cli_par_fib", "%d", "%d %d")

    # Send all of the requests to the server
    for i in range(1, iterations + 1):
        print('rpcFibCliPar(P): fib(->%-2d) ::: ' % i)
        ret = rpc.rpcSend(i)
        if not ret:
            print('rpcFibCliPar(P): SEND FAILED')
            sys.exit(-1)

    # Receive responses for all requests that were sent
    for i in range(1, iterations + 1):
        ret, fib = rpc.rpcRecv()
        if not ret:
            print('rpcFibCliPar(P): RECV FAILED')
            break
        print('rpcFibCliPar(P): fib(%2d<-) = %-2d<-' % fib)

    print('Goodbye from Python rpcFibCliPar')
    sys.exit(0)

    
if __name__ == '__main__':
    fibClient(sys.argv[1:])

File: cis_interface/examples/rpcFib/src/rpcFibCliPar.c

#include "PsiInterface.h"
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <signal.h>


int main(int argc, char *argv[]) {

  int iterations = atoi(argv[1]);
  printf("Hello from C rpcFibCliPar: iterations = %d\n", iterations);
  
  // Create RPC connection with server 
  psiRpc_t rpc = psiRpcClient("cli_par_fib", "%d", "%d %d");
  
  // Send all of the requests to the server
  int ret;
  for (int i = 1; i <= iterations; i++) {
    printf("rpcFibCliPar(C): fib(->%-2d) ::: \n", i);
    ret = rpcSend(rpc, i);
    if (ret != 0) {
      printf("rpcFibCliPar(C): SEND FAILED\n");
      exit(-1);
    }
  }

  // Receive responses for all requests that were sent
  int fib = -1;
  int fibNo = -1;
  for (int i = 1; i <= iterations; i++) {
    ret = rpcRecv(rpc, &fibNo, &fib);
    if (ret < 0) {
      printf("rpcFibCliPar(C): RECV FAILED\n");
      exit(-1);
    }
    printf("rpcFibCliPar(C):  fib(%2d<-) = %-2d<-\n", fibNo, fib);
  }

  printf("Goodbye from C rpcFibCliPar\n");
  exit(0);
    
}

